ldap/3.4/modules/ROOT/pages/ldap-connector-reference.adoc

```diff
@@ -0,0 +1,1105 @@
+|Show attributes as Objects | boolean|If set to true, the queries display the entity attributes as objects, containing the definition type, a flag to indicate wether or not the attribute is multi-value, and the value of the attribute. If set to false, the queries only display the attribute values. |false ||
+|===
+
+== Basic Configuration Type
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Principal DN a| String |  |  | x
+| Password a| String |  |  |
+| Authentication a| String |  |  |
+| URL a| String |  |  | x
+| Reconnection a| Reconnection |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
+| Type | Enumeration, one of:
+
+** JNDI |  The implementation of the connection to use. |  JNDI |
+| Schema Enabled a| Boolean |  If set to true, the LDAP connector uses the LDAP schema (only works for LDAP v3) to define the structure of the LDAP entry or map. This must be 'true' to use DataSense because it affects the implementing class of org.mule.module.ldap.api.LDAPEntry attributes. |  false |
+| Referral a| Enumeration, one of:
+
+** IGNORE
+** THROW
+** FOLLOW |  Constant that holds the name of the environment property for specifying how referrals encountered by the service provider are to be processed (follow, ignore, throw). |  IGNORE |
+| Extended Configuration a| Object |  This is a Map instance holding extended configuration attributes to use in the Context environment. When working with TLS connections make sure that the native LDAP pooling functionality is turned off. For example, if using JNDI, do not use attributes such as `com.sun.jndi.ldap.connect.pool=true`, which causes problems when using TLS. |  |
+| Initial Pool Size a| Number |  The string representation of an integer that represents the number of connections per connection identity to create when initially creating a connection for the identity. To disable pooling, just set this value to 0 (zero). |  1 |
+| Max Pool Size a| Number |  The string representation of an integer that represents the maximum number of connections per connection identity that can be maintained concurrently. |  5 |
+| Pool Timeout a| Number |  The string representation of an integer that represents the number of milliseconds that an idle connection may remain in the pool without being closed and removed from the pool. |  60000 |
+|===
+
+== SSL Configuration
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Principal DN a| String |  |  | x
```

Comment:
All of these fields should have descriptions. This is an issue with the java files. We should open a ticket against the connector to add descriptions for these fields. You can add them manually if you have time for this release but then we would have to get a technical review. Like, for Principal DN, does it accept either the name of the domain OR the IP address? Or just the name? 

Principal DN: Name or IP address of the domain used by your LDAP implementation.

Reply:
Actually, I don't even know for sure that DN is referring to domain name.